pkg/controller/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,405 @@
+	n.enqueueInternalNetworkPolicy(getBANPReference(curBANP))
+}
+
+// deleteBANP receives BaselineAdminNetworkPolicy DELETE events and enqueues a
+// reference of the BaselineAdminNetworkPolicy to trigger its process.
+func (n *NetworkPolicyController) deleteBANP(old interface{}) {
+	banp, ok := old.(*v1alpha1.BaselineAdminNetworkPolicy)
+	if !ok {
+		tombstone, ok := old.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("Error decoding object when deleting BaselineAdminNetworkPolicy, invalid type: %v", old)
+			return
+		}
+		banp, ok = tombstone.Obj.(*v1alpha1.BaselineAdminNetworkPolicy)
+		if !ok {
+			klog.Errorf("Error decoding object tombstone when deleting BaselineAdminNetworkPolicy, invalid type: %v", tombstone.Obj)
+			return
+		}
+	}
+	defer n.heartbeat("deleteBANP")
+	klog.InfoS("Processing BaselineAdminNetworkPolicy DELETE event", "banp", banp.Name)
+	n.enqueueInternalNetworkPolicy(getBANPReference(banp))
+}
+
+// anpHasNamespaceLabelRule returns whether an AdminNetworkPolicy has rules defined by
+// advanced Namespace selection (sameLabels and notSameLabels)
+func anpHasNamespaceLabelRule(anp *v1alpha1.AdminNetworkPolicy) bool {
+	for _, ingress := range anp.Spec.Ingress {
+		for _, peer := range ingress.From {
+			if peer.Namespaces != nil && (len(peer.Namespaces.SameLabels) > 0 || len(peer.Namespaces.NotSameLabels) > 0) {
+				return true
+			}
+		}
+	}
+	for _, egress := range anp.Spec.Egress {
+		for _, peer := range egress.To {
+			if peer.Namespaces != nil && (len(peer.Namespaces.SameLabels) > 0 || len(peer.Namespaces.NotSameLabels) > 0) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// banpHasNamespaceLabelRule returns whether a BaselineAdminNetworkPolicy has rules defined by
+// advanced Namespace selection (sameLabels and notSameLabels)
+func banpHasNamespaceLabelRule(banp *v1alpha1.BaselineAdminNetworkPolicy) bool {
+	for _, ingress := range banp.Spec.Ingress {
+		for _, peer := range ingress.From {
+			if peer.Namespaces != nil && (len(peer.Namespaces.SameLabels) > 0 || len(peer.Namespaces.NotSameLabels) > 0) {
+				return true
+			}
+		}
+	}
+	for _, egress := range banp.Spec.Egress {
+		for _, peer := range egress.To {
+			if peer.Namespaces != nil && (len(peer.Namespaces.SameLabels) > 0 || len(peer.Namespaces.NotSameLabels) > 0) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// toAntreaServicesForPolicyCRD processes ports field for ANPs/BANPs and returns the translated
+// Antrea Services.
+func toAntreaServicesForPolicyCRD(npPorts []v1alpha1.AdminNetworkPolicyPort) []controlplane.Service {
+	var antreaServices []controlplane.Service
+	for _, npPort := range npPorts {
+		if npPort.PortNumber != nil {
+			port := intstr.FromInt(int(npPort.PortNumber.Port))
+			antreaServices = append(antreaServices, controlplane.Service{
+				Protocol: toAntreaProtocol(&npPort.PortNumber.Protocol),
+				Port:     &port,
+			})
+		} else if npPort.PortRange != nil {
```

Comment:
Are they (`PortNumber` `PortRange` and `NamedPort`) exclusive? I suppose you can remove `else`?

Reply:
These are actually exclusive, as defined by API spec